### PR TITLE
Add build for osx_arm64

### PR DESCRIPTION
**Pull request by mmcauliffe**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -3,11 +3,18 @@
...
+
 ./configure \
    --prefix="${PREFIX}" \
    --enable-compress \
    --enable-fsts \
    --enable-grm \
    --enable-special
-make -j"${CPU_COUNT}" check
+   
+if [[ "$CONDA_BUILD_CROSS_COMPILATION" != "1" ]]; then
```
Comment (kylebgorman) [inline recipe/build.sh:22]: I am not sure what the connection between x-compile and running `check` is: could you add a comment?

**Comment by kylebgorman** [comment]
Hey, funny seeing you here.

**Comment by mmcauliffe** [comment]
Well hello! I've gotten a few issues about osx_arm64 not working for MFA, and I was hoping getting openfst (and maybe the rest of the pynini flow) working would be simple, but I'm hitting an issue in configuration.

Was about to file an issue, but maybe this is easier, is it even possible currently to cross compile with openfst? I'm hitting the float equality test block, but it doesn't look like it's skippable through configuration options (with my admittedly limited knowledge of configure and make in general).  Would it be ok to do a osx_arm64 only patch that removes that block?

**Comment by kylebgorman** [comment]
> Well hello! I've gotten a few issues about osx_arm64 not working for MFA, and I was hoping getting openfst (and maybe the rest of the pynini flow) working would be simple, but I'm hitting an issue in configuration.
> 
> Was about to file an issue, but maybe this is easier, is it even possible currently to cross compile with openfst? I'm hitting the float equality test block, but it doesn't look like it's skippable through configuration options (with my admittedly limited knowledge of configure and make in general). Would it be ok to do a osx_arm64 only patch that removes that block?

Have never tried to cross-compile really anything. 

But I'm not opposed. We certainly should support the new Macs. Question is: does the patch live here in Conda-Forge or go upstream?

Haven't looked into how to make an assertion test skippable like that.

**Comment by kylebgorman** [comment]
Patch here seems fine to me if tests pass. We can get rid of it once native osx-arm64 compilation capacities are available on Azure and/or enabled (whatever is necessary to make x-compile unnecessary on that platform).

**Comment by mmcauliffe** [comment]
Ok, sounds good! I'm trying out just a patch that doesn't have `as_fn_error` in the `"$cross_compiling" = yes` block of that test, so this is pretty hacky and I don't know if it would be good in the upstream.  I also don't know if float equality is a major concern for cross compiling in general. The error message for it when not cross compiling is to include `-msse -mfpmath=sse`, so that might work when cross compiling too.

**Comment by mmcauliffe** [comment]
Looks like everything's building now!

File: recipe/build.sh (hunk 1 above)
Comment (kylebgorman) [reply recipe/build.sh:22]: Is this stll necessary?

File: recipe/build.sh [hunk 2]
```diff
@@ -3,11 +3,25 @@
 # https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
 export CXXFLAGS="$(echo "${CXXFLAGS}" | sed -E 's@-std=c\+\+[^ ]+@@g') -D_LIBCPP_DISABLE_AVAILABILITY"
 
+# Get an updated config.sub and config.guess
+cp $BUILD_PREFIX/share/gnuconfig/config.* .
+
+if [[ "${CONDA_BUILD_CROSS_COMPILATION}" == "1" ]]; then
+  OPENFST_CROSS_COMPILATION_CONFIGURE_OPTS="--build=${BUILD} --host=${HOST}"
+  echo "Adding cross compile options: ${OPENFST_CROSS_COMPILATION_CONFIGURE_OPTS}"
```
Comment (kylebgorman) [inline recipe/build.sh]: where does this echo go?

**Review by kylebgorman — APPROVED** [review]
Two Q's but go ahead; LGTM

File: recipe/build.sh (hunk 2 above)
Comment (mmcauliffe) [reply recipe/build.sh]: Oh, just a debug message, removed it in the latest commit.

File: recipe/build.sh (hunk 1 above)
Comment (mmcauliffe) [reply recipe/build.sh:22]: I think so, these lines are from https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/#how-to-add-a-osx-arm64-build-to-a-feedstock

**Comment by mmcauliffe** [comment]
@kylebgorman ok, should be good for you to merge in now

**Pull request merged**